jvm/src/test/scala/org/apache/spark/sql/GlutenQueryTest.scala

```diff
@@ -0,0 +1,437 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+/**
+ * Why we need a GlutenQueryTest when we already have QueryTest?
+ * 1. We need to modify the way org.apache.spark.sql.CHQueryTest#compare compares double
+ */
+import org.apache.spark.sql.catalyst.plans._
+import org.apache.spark.sql.catalyst.util._
+import org.apache.spark.sql.execution.SQLExecution
+import org.apache.spark.sql.execution.columnar.InMemoryRelation
+import org.apache.spark.storage.StorageLevel
+import org.junit.Assert
+import org.scalatest.Assertions
+
+import java.util.TimeZone
+import scala.collection.JavaConverters._
+
+
+abstract class GlutenQueryTest extends PlanTest {
```

Comment:
Is there any reason not to utilize `org.apache.spark.sql.QueryTest` directly?

Reply:
It needs to modify the `compare` method of QueryTest, when comparing double value, there is some different between the result of native engine and the one of Vanilla Spark.

Reply:
I notice both `org.apache.spark.sql.QueryTest#checkDataset` and `org.apache.spark.sql.QueryTest#checkDatasetUnorderly` are marked protected. Maybe it is worth to try simplify codes by overriding the methods?

Reply:
It occurs to me that `org.apache.spark.sql.QueryTest#getResult` is private too. Let's just keep the copied codes if things are getting complicated.

Reply:
The method `compare` is private.